### ISSUE TITLE
Add research for <dialog> and related popups

### ISSUE THREAD

**Issue by melanierichards** [openui/open-ui#116]
Currently conducting some research on various flavors of popups, in order to determine how we might make these easier to author with the right accessibility semantics, user input behaviors, styling, etc. Findings from this research should be added to the Open UI project. :)

**Comment by melanierichards** [comment]
@gregwhitworth I can't self-assign, mind assigning me to this issue?

**Comment by melanierichards** [comment]
PR #205 was opened to address this, and was intentionally left very broad. I'd like to split up this PR into "things we think are a good match for `<popup>`" and "other top layer things". Per [MS Edge Explainers #455](https://github.com/MicrosoftEdge/MSEdgeExplainers/issues/455) we should audit various ARIA roles and make sure we've got all the popup-y things. That was part of my initial research, but it's been a bit so I'll go back with a fine-toothed comb and make sure I didn't miss something.

**Comment by andrico1234** [comment]
I'll close this ticket as the popup API spec has recently been merged into the [Open UI site](https://open-ui.org/components/popup.research.explainer) and [this issue](https://github.com/openui/open-ui/issues/18) focuses specifically on a `<dialog>` spec.

@melanierichards  Feel free to reopen if I've gotten too trigger happy with the close button 😄
